addons/account/models/account_move.py

```diff
@@ -6354,12 +6354,12 @@ def _get_invoice_legal_documents_all(self, allow_fallback=False):
         elif allow_fallback:
             return [self._get_invoice_pdf_proforma()]
 
-    def _get_invoice_report_filename(self, extension='pdf'):
+    def _get_invoice_report_filename(self, extension='pdf', report=None):
         """ Get the filename of the generated invoice report with extension file. """
         self.ensure_one()
-        report_id = self.partner_id.invoice_template_pdf_report_id or self.env.ref('account.account_invoices')
+        report_id = report or self.env['account.move.send']._get_default_pdf_report_id(self)
```

Comment:
rename `report_id` to `report` plz. It's confusing if it's a record or an id.

Reply:
By design, this is really bad... The docstring doesn't say it can only be called when using the send & print wizard. I'd prefer:
```suggestion
        if not report:
            report = self.partner_id.invoice_template_pdf_report_id or self.env.ref('account.account_invoices')
```
... but make sure the parameter is always given by the send & print wizard